### PR TITLE
Rustfmt

### PR DESCRIPTION
**Pull request by posix4e**
If we want to keep everything in line with rustfmt, this would be a way to do it.

### CODE REVIEW

**Comment by posix4e** [comment]
Also this patch is trivial for me to rebase so feel free to merge others and I will be glad to up-to-date-the-patch

File: bin/format_code.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+for i in `find src -type f -name '*.rs'`
+	do rustfmt --write-mode=overwrite $i
+done
```
Comment (tschottdorf) [inline bin/format_code.sh]: for handling whitespace, etc correctly:
```bash
find src -type f -name '*.rs' -print0 | while read -d $'\0' -r f; do
  rustfmt --write-mode=overwrite "${f}"
done
```
not that we're likely to need this here, but still.

File: default.toml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+struct_trailing_comma = "Vertical"
+struct_lit_style = "BlockIndent"
+struct_lit_trailing_comma = "Vertical"
+enum_trailing_comma = true
+report_todo = "Always"
+report_fixme = "Never"
+reorder_imports = false
+expr_indent_style = "Tabbed"
+closure_indent_style = "Visual"
+single_line_if_else = false
```
Comment (tschottdorf) [inline default.toml]: are the settings in this file taken from a "standard" defaults or based on anything?

**Comment by tschottdorf** [comment]
:+1:, though I don't see the value in having a commit for each individual file changed. As far as I'm concerned, it would make sense to squash them.

File: src/connection.rs [hunk 3]
```diff
@@ -5,24 +5,9 @@ use std::rc::Rc;
 
 use mio::tcp::TcpStream;
 use mio::Timeout as TimeoutHandle;
-use mio::{
```
Comment (Hoverbear) [inline src/connection.rs]: I feel like this style of importing is much much easier to read.

File: default.toml (hunk 2 above)
Comment (posix4e) [inline default.toml]: These are the same defaults as what rustfmt uses on itself.

On Wed, Aug 26, 2015, 22:31 Tobias Schottdorf <notifications@github.com>
wrote:

> In default.toml
> <https://github.com/Hoverbear/raft-rs/pull/108#discussion_r38063408>:
>
> > +leeway = 5
> > +tab_spaces = 4
> > +newline_style = "Unix"
> > +fn_brace_style = "SameLineWhere"
> > +fn_return_indent = "WithArgs"
> > +fn_args_paren_newline = true
> > +struct_trailing_comma = "Vertical"
> > +struct_lit_style = "BlockIndent"
> > +struct_lit_trailing_comma = "Vertical"
> > +enum_trailing_comma = true
> > +report_todo = "Always"
> > +report_fixme = "Never"
> > +reorder_imports = false
> > +expr_indent_style = "Tabbed"
> > +closure_indent_style = "Visual"
> > +single_line_if_else = false
>
> are the settings in this file taken from a "standard" defaults or based on
> anything?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Hoverbear/raft-rs/pull/108/files#r38063408>.
>
-- 

Sent from my mobile 5045076749
Alex Newman

File: src/connection.rs (hunk 3 above)
Comment (tschottdorf) [inline src/connection.rs]: Takes a lot of space for something you barely ever read and which tooling manages for you (ideally). I'm happy having it collapsed.

**Comment by posix4e** [comment]
I just combined all the commits. Anything else we want to see?

File: src/connection.rs (hunk 3 above)
Comment (posix4e) [inline src/connection.rs]: So you don't like the collapse? I wonder if that's something I can change about rustfmt

File: bin/format_code.sh (hunk 1 above)
Comment (posix4e) [inline bin/format_code.sh]: For some reason that doesn't seem to work on my computer. Perhaps a second commit?

**Comment by posix4e** [comment]
Dan pointed out examples should be formatted as well. I made that change

**Comment by posix4e** [comment]
I plopped a patch on rustfmt to not try to align strings which the user already has newlines
https://github.com/nrc/rustfmt/pull/235. I reformatted it using the version in that pull request

File: rustfmt.toml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+newline_style = "Unix"
+fn_brace_style = "SameLineWhere"
+fn_return_indent = "WithArgs"
+fn_args_paren_newline = true
+struct_trailing_comma = "Vertical"
+struct_lit_style = "BlockIndent"
+struct_lit_trailing_comma = "Vertical"
+enum_trailing_comma = true
+report_todo = "Always"
+report_fixme = "Never"
```
Comment (Hoverbear) [inline rustfmt.toml]: If we're reporting `TODO:` we should probably do `FIXME:` too.

File: src/connection.rs (hunk 3 above)
Comment (Hoverbear) [inline src/connection.rs]: Yeah. it makes it really hard to enumerate them visually. The [style guide](http://doc.rust-lang.org/nightly/style/style/imports.html) doesn't seem to offer much guidance in this though.

File: src/state_machine/channel.rs [hunk 5]
```diff
@@ -8,7 +8,7 @@ use state_machine::StateMachine;
 ///
 /// This state machine is chiefly meant for testing.
 pub struct ChannelStateMachine {
-    tx: mpsc::Sender<Vec<u8>>
+    tx: mpsc::Sender<Vec<u8>>,
```
Comment (Hoverbear) [inline src/state_machine/channel.rs]: Oh trailing commas, how I love thee. :(

**Comment by posix4e** [comment]
Upon looking into this futher I think we need a bunch of improvements to rustfmt before we can take it as gospel. I'll close this for now

**Comment by Hoverbear** [comment]
@posix4e Let's leave this open, actually! I'd like to ~one day~ do it.

**Comment by posix4e** [comment]
Actually It seems as though I was able to get rid of the trailing commas and prevent the examples from getting mangled. I turned on reordering of imports. I think it reads much easier that way, but it'll create a lot of churn. The main thing that worried me was the panicing.

**Comment by Hoverbear** [comment]
There has been many updates to `rustfmt` and it might be good to take a new look at this!

**Comment by tschottdorf** [comment]
+1

**Comment by Hoverbear** [comment]
You can now run `cargo format` with Rustfmt!

**Comment by posix4e** [comment]
hah!

**Pull request closed without merging**